// @openrouter/sdk/esm/react-query/analyticsGetUserActivity.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { analyticsGetUserActivity } from "../funcs/analyticsGetUserActivity.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Get user activity grouped by endpoint
 *
 * @remarks
 * Returns user activity data grouped by endpoint for the last 30 (completed) UTC days
 */
export function useAnalyticsGetUserActivity(request, options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildAnalyticsGetUserActivityQuery(client, request, options),
        ...options,
    });
}
/**
 * Get user activity grouped by endpoint
 *
 * @remarks
 * Returns user activity data grouped by endpoint for the last 30 (completed) UTC days
 */
export function useAnalyticsGetUserActivitySuspense(request, options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildAnalyticsGetUserActivityQuery(client, request, options),
        ...options,
    });
}
export function prefetchAnalyticsGetUserActivity(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildAnalyticsGetUserActivityQuery(client$, request),
    });
}
export function setAnalyticsGetUserActivityData(client, queryKeyBase, data) {
    const key = queryKeyAnalyticsGetUserActivity(...queryKeyBase);
    return client.setQueryData(key, data);
}
export function invalidateAnalyticsGetUserActivity(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@openrouter/sdk",
            "Analytics",
            "getUserActivity",
            ...queryKeyBase,
        ],
    });
}
export function invalidateAllAnalyticsGetUserActivity(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Analytics", "getUserActivity"],
    });
}
export function buildAnalyticsGetUserActivityQuery(client$, request, options) {
    return {
        queryKey: queryKeyAnalyticsGetUserActivity({ date: request?.date }),
        queryFn: async function analyticsGetUserActivityQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(analyticsGetUserActivity(client$, request, mergedOptions));
        },
    };
}
export function queryKeyAnalyticsGetUserActivity(parameters) {
    return ["@openrouter/sdk", "Analytics", "getUserActivity", parameters];
}
