// @openrouter/sdk/esm/react-query/apiKeysGet.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { apiKeysGet } from "../funcs/apiKeysGet.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Get a single API key
 */
export function useApiKeysGet(request, options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildApiKeysGetQuery(client, request, options),
        ...options,
    });
}
/**
 * Get a single API key
 */
export function useApiKeysGetSuspense(request, options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildApiKeysGetQuery(client, request, options),
        ...options,
    });
}
export function prefetchApiKeysGet(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildApiKeysGetQuery(client$, request),
    });
}
export function setApiKeysGetData(client, queryKeyBase, data) {
    const key = queryKeyApiKeysGet(...queryKeyBase);
    return client.setQueryData(key, data);
}
export function invalidateApiKeysGet(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "API Keys", "get", ...queryKeyBase],
    });
}
export function invalidateAllApiKeysGet(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "API Keys", "get"],
    });
}
export function buildApiKeysGetQuery(client$, request, options) {
    return {
        queryKey: queryKeyApiKeysGet(request.hash),
        queryFn: async function apiKeysGetQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(apiKeysGet(client$, request, mergedOptions));
        },
    };
}
export function queryKeyApiKeysGet(hash) {
    return ["@openrouter/sdk", "API Keys", "get", hash];
}
