// @openrouter/sdk/esm/react-query/apiKeysCreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { apiKeysCreate } from "../funcs/apiKeysCreate.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Create a new API key
 */
export function useApiKeysCreateMutation(options) {
    const client = useOpenRouterContext();
    return useMutation({
        ...buildApiKeysCreateMutation(client, options),
        ...options,
    });
}
export function mutationKeyApiKeysCreate() {
    return ["@openrouter/sdk", "API Keys", "create"];
}
export function buildApiKeysCreateMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyApiKeysCreate(),
        mutationFn: function apiKeysCreateMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(apiKeysCreate(client$, request, mergedOptions));
        },
    };
}
