// @openrouter/sdk/esm/react-query/apiKeysUpdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { apiKeysUpdate } from "../funcs/apiKeysUpdate.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Update an API key
 */
export function useApiKeysUpdateMutation(options) {
    const client = useOpenRouterContext();
    return useMutation({
        ...buildApiKeysUpdateMutation(client, options),
        ...options,
    });
}
export function mutationKeyApiKeysUpdate() {
    return ["@openrouter/sdk", "API Keys", "update"];
}
export function buildApiKeysUpdateMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyApiKeysUpdate(),
        mutationFn: function apiKeysUpdateMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(apiKeysUpdate(client$, request, mergedOptions));
        },
    };
}
