// @openrouter/sdk/esm/react-query/apiKeysGetCurrentKeyMetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { apiKeysGetCurrentKeyMetadata } from "../funcs/apiKeysGetCurrentKeyMetadata.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Get current API key
 *
 * @remarks
 * Get information on the API key associated with the current authentication session
 */
export function useApiKeysGetCurrentKeyMetadata(options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildApiKeysGetCurrentKeyMetadataQuery(client, options),
        ...options,
    });
}
/**
 * Get current API key
 *
 * @remarks
 * Get information on the API key associated with the current authentication session
 */
export function useApiKeysGetCurrentKeyMetadataSuspense(options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildApiKeysGetCurrentKeyMetadataQuery(client, options),
        ...options,
    });
}
export function prefetchApiKeysGetCurrentKeyMetadata(queryClient, client$) {
    return queryClient.prefetchQuery({
        ...buildApiKeysGetCurrentKeyMetadataQuery(client$),
    });
}
export function setApiKeysGetCurrentKeyMetadataData(client, data) {
    const key = queryKeyApiKeysGetCurrentKeyMetadata();
    return client.setQueryData(key, data);
}
export function invalidateAllApiKeysGetCurrentKeyMetadata(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "API Keys", "getCurrentKeyMetadata"],
    });
}
export function buildApiKeysGetCurrentKeyMetadataQuery(client$, options) {
    return {
        queryKey: queryKeyApiKeysGetCurrentKeyMetadata(),
        queryFn: async function apiKeysGetCurrentKeyMetadataQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(apiKeysGetCurrentKeyMetadata(client$, mergedOptions));
        },
    };
}
export function queryKeyApiKeysGetCurrentKeyMetadata() {
    return ["@openrouter/sdk", "API Keys", "getCurrentKeyMetadata"];
}
