// @openrouter/sdk/esm/react-query/apiKeysDelete.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { apiKeysDelete } from "../funcs/apiKeysDelete.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Delete an API key
 */
export function useApiKeysDeleteMutation(options) {
    const client = useOpenRouterContext();
    return useMutation({
        ...buildApiKeysDeleteMutation(client, options),
        ...options,
    });
}
export function mutationKeyApiKeysDelete() {
    return ["@openrouter/sdk", "API Keys", "delete"];
}
export function buildApiKeysDeleteMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyApiKeysDelete(),
        mutationFn: function apiKeysDeleteMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(apiKeysDelete(client$, request, mergedOptions));
        },
    };
}
