// @openrouter/sdk/esm/react-query/apiKeysList.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { apiKeysList } from "../funcs/apiKeysList.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * List API keys
 */
export function useApiKeysList(request, options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildApiKeysListQuery(client, request, options),
        ...options,
    });
}
/**
 * List API keys
 */
export function useApiKeysListSuspense(request, options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildApiKeysListQuery(client, request, options),
        ...options,
    });
}
export function prefetchApiKeysList(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildApiKeysListQuery(client$, request),
    });
}
export function setApiKeysListData(client, queryKeyBase, data) {
    const key = queryKeyApiKeysList(...queryKeyBase);
    return client.setQueryData(key, data);
}
export function invalidateApiKeysList(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "API Keys", "list", ...queryKeyBase],
    });
}
export function invalidateAllApiKeysList(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "API Keys", "list"],
    });
}
export function buildApiKeysListQuery(client$, request, options) {
    return {
        queryKey: queryKeyApiKeysList({
            includeDisabled: request?.includeDisabled,
            offset: request?.offset,
        }),
        queryFn: async function apiKeysListQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(apiKeysList(client$, request, mergedOptions));
        },
    };
}
export function queryKeyApiKeysList(parameters) {
    return ["@openrouter/sdk", "API Keys", "list", parameters];
}
