// @openrouter/sdk/esm/react-query/betaResponsesSend.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { betaResponsesSend } from "../funcs/betaResponsesSend.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Create a response
 *
 * @remarks
 * Creates a streaming or non-streaming response using OpenResponses API format
 */
export function useBetaResponsesSendMutation(options) {
    const client = useOpenRouterContext();
    return useMutation({
        ...buildBetaResponsesSendMutation(client, options),
        ...options,
    });
}
export function mutationKeyBetaResponsesSend() {
    return ["@openrouter/sdk", "responses", "send"];
}
export function buildBetaResponsesSendMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyBetaResponsesSend(),
        mutationFn: function betaResponsesSendMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(betaResponsesSend(client$, request, mergedOptions));
        },
    };
}
