// @openrouter/sdk/esm/react-query/chatSend.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { chatSend } from "../funcs/chatSend.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Create a chat completion
 *
 * @remarks
 * Sends a request for a model response for the given chat conversation. Supports both streaming and non-streaming modes.
 */
export function useChatSendMutation(options) {
    const client = useOpenRouterContext();
    return useMutation({
        ...buildChatSendMutation(client, options),
        ...options,
    });
}
export function mutationKeyChatSend() {
    return ["@openrouter/sdk", "chat", "send"];
}
export function buildChatSendMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyChatSend(),
        mutationFn: function chatSendMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(chatSend(client$, request, mergedOptions));
        },
    };
}
