// @openrouter/sdk/esm/react-query/completionsGenerate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { completionsGenerate } from "../funcs/completionsGenerate.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Create a completion
 *
 * @remarks
 * Creates a completion for the provided prompt and parameters. Supports both streaming and non-streaming modes.
 */
export function useCompletionsGenerateMutation(options) {
    const client = useOpenRouterContext();
    return useMutation({
        ...buildCompletionsGenerateMutation(client, options),
        ...options,
    });
}
export function mutationKeyCompletionsGenerate() {
    return ["@openrouter/sdk", "completions", "generate"];
}
export function buildCompletionsGenerateMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyCompletionsGenerate(),
        mutationFn: function completionsGenerateMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(completionsGenerate(client$, request, mergedOptions));
        },
    };
}
