// @openrouter/sdk/esm/react-query/creditsCreateCoinbaseCharge.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { creditsCreateCoinbaseCharge } from "../funcs/creditsCreateCoinbaseCharge.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Create a Coinbase charge for crypto payment
 *
 * @remarks
 * Create a Coinbase charge for crypto payment
 */
export function useCreditsCreateCoinbaseChargeMutation(options) {
    const client = useOpenRouterContext();
    return useMutation({
        ...buildCreditsCreateCoinbaseChargeMutation(client, options),
        ...options,
    });
}
export function mutationKeyCreditsCreateCoinbaseCharge() {
    return ["@openrouter/sdk", "Credits", "createCoinbaseCharge"];
}
export function buildCreditsCreateCoinbaseChargeMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyCreditsCreateCoinbaseCharge(),
        mutationFn: function creditsCreateCoinbaseChargeMutationFn({ security, request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(creditsCreateCoinbaseCharge(client$, security, request, mergedOptions));
        },
    };
}
