// @openrouter/sdk/esm/react-query/creditsGetCredits.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { creditsGetCredits } from "../funcs/creditsGetCredits.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Get remaining credits
 *
 * @remarks
 * Get total credits purchased and used for the authenticated user
 */
export function useCreditsGetCredits(options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildCreditsGetCreditsQuery(client, options),
        ...options,
    });
}
/**
 * Get remaining credits
 *
 * @remarks
 * Get total credits purchased and used for the authenticated user
 */
export function useCreditsGetCreditsSuspense(options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildCreditsGetCreditsQuery(client, options),
        ...options,
    });
}
export function prefetchCreditsGetCredits(queryClient, client$) {
    return queryClient.prefetchQuery({
        ...buildCreditsGetCreditsQuery(client$),
    });
}
export function setCreditsGetCreditsData(client, data) {
    const key = queryKeyCreditsGetCredits();
    return client.setQueryData(key, data);
}
export function invalidateAllCreditsGetCredits(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Credits", "getCredits"],
    });
}
export function buildCreditsGetCreditsQuery(client$, options) {
    return {
        queryKey: queryKeyCreditsGetCredits(),
        queryFn: async function creditsGetCreditsQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(creditsGetCredits(client$, mergedOptions));
        },
    };
}
export function queryKeyCreditsGetCredits() {
    return ["@openrouter/sdk", "Credits", "getCredits"];
}
