// @openrouter/sdk/esm/react-query/embeddingsListModels.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { embeddingsListModels } from "../funcs/embeddingsListModels.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * List all embeddings models
 *
 * @remarks
 * Returns a list of all available embeddings models and their properties
 */
export function useEmbeddingsListModels(options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildEmbeddingsListModelsQuery(client, options),
        ...options,
    });
}
/**
 * List all embeddings models
 *
 * @remarks
 * Returns a list of all available embeddings models and their properties
 */
export function useEmbeddingsListModelsSuspense(options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildEmbeddingsListModelsQuery(client, options),
        ...options,
    });
}
export function prefetchEmbeddingsListModels(queryClient, client$) {
    return queryClient.prefetchQuery({
        ...buildEmbeddingsListModelsQuery(client$),
    });
}
export function setEmbeddingsListModelsData(client, data) {
    const key = queryKeyEmbeddingsListModels();
    return client.setQueryData(key, data);
}
export function invalidateAllEmbeddingsListModels(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Embeddings", "listModels"],
    });
}
export function buildEmbeddingsListModelsQuery(client$, options) {
    return {
        queryKey: queryKeyEmbeddingsListModels(),
        queryFn: async function embeddingsListModelsQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(embeddingsListModels(client$, mergedOptions));
        },
    };
}
export function queryKeyEmbeddingsListModels() {
    return ["@openrouter/sdk", "Embeddings", "listModels"];
}
