// @openrouter/sdk/esm/react-query/embeddingsGenerate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { embeddingsGenerate } from "../funcs/embeddingsGenerate.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Submit an embedding request
 *
 * @remarks
 * Submits an embedding request to the embeddings router
 */
export function useEmbeddingsGenerateMutation(options) {
    const client = useOpenRouterContext();
    return useMutation({
        ...buildEmbeddingsGenerateMutation(client, options),
        ...options,
    });
}
export function mutationKeyEmbeddingsGenerate() {
    return ["@openrouter/sdk", "Embeddings", "generate"];
}
export function buildEmbeddingsGenerateMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyEmbeddingsGenerate(),
        mutationFn: function embeddingsGenerateMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(embeddingsGenerate(client$, request, mergedOptions));
        },
    };
}
