// @openrouter/sdk/esm/react-query/endpointsListZdrEndpoints.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { endpointsListZdrEndpoints } from "../funcs/endpointsListZdrEndpoints.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Preview the impact of ZDR on the available endpoints
 */
export function useEndpointsListZdrEndpoints(options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildEndpointsListZdrEndpointsQuery(client, options),
        ...options,
    });
}
/**
 * Preview the impact of ZDR on the available endpoints
 */
export function useEndpointsListZdrEndpointsSuspense(options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildEndpointsListZdrEndpointsQuery(client, options),
        ...options,
    });
}
export function prefetchEndpointsListZdrEndpoints(queryClient, client$) {
    return queryClient.prefetchQuery({
        ...buildEndpointsListZdrEndpointsQuery(client$),
    });
}
export function setEndpointsListZdrEndpointsData(client, data) {
    const key = queryKeyEndpointsListZdrEndpoints();
    return client.setQueryData(key, data);
}
export function invalidateAllEndpointsListZdrEndpoints(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Endpoints", "listZdrEndpoints"],
    });
}
export function buildEndpointsListZdrEndpointsQuery(client$, options) {
    return {
        queryKey: queryKeyEndpointsListZdrEndpoints(),
        queryFn: async function endpointsListZdrEndpointsQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(endpointsListZdrEndpoints(client$, mergedOptions));
        },
    };
}
export function queryKeyEndpointsListZdrEndpoints() {
    return ["@openrouter/sdk", "Endpoints", "listZdrEndpoints"];
}
