// @openrouter/sdk/esm/react-query/endpointsList.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { endpointsList } from "../funcs/endpointsList.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * List all endpoints for a model
 */
export function useEndpointsList(request, options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildEndpointsListQuery(client, request, options),
        ...options,
    });
}
/**
 * List all endpoints for a model
 */
export function useEndpointsListSuspense(request, options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildEndpointsListQuery(client, request, options),
        ...options,
    });
}
export function prefetchEndpointsList(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildEndpointsListQuery(client$, request),
    });
}
export function setEndpointsListData(client, queryKeyBase, data) {
    const key = queryKeyEndpointsList(...queryKeyBase);
    return client.setQueryData(key, data);
}
export function invalidateEndpointsList(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Endpoints", "list", ...queryKeyBase],
    });
}
export function invalidateAllEndpointsList(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Endpoints", "list"],
    });
}
export function buildEndpointsListQuery(client$, request, options) {
    return {
        queryKey: queryKeyEndpointsList(request.author, request.slug),
        queryFn: async function endpointsListQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(endpointsList(client$, request, mergedOptions));
        },
    };
}
export function queryKeyEndpointsList(author, slug) {
    return ["@openrouter/sdk", "Endpoints", "list", author, slug];
}
