// @openrouter/sdk/esm/react-query/generationsGetGeneration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { generationsGetGeneration } from "../funcs/generationsGetGeneration.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Get request & usage metadata for a generation
 */
export function useGenerationsGetGeneration(request, options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildGenerationsGetGenerationQuery(client, request, options),
        ...options,
    });
}
/**
 * Get request & usage metadata for a generation
 */
export function useGenerationsGetGenerationSuspense(request, options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildGenerationsGetGenerationQuery(client, request, options),
        ...options,
    });
}
export function prefetchGenerationsGetGeneration(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildGenerationsGetGenerationQuery(client$, request),
    });
}
export function setGenerationsGetGenerationData(client, queryKeyBase, data) {
    const key = queryKeyGenerationsGetGeneration(...queryKeyBase);
    return client.setQueryData(key, data);
}
export function invalidateGenerationsGetGeneration(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@openrouter/sdk",
            "Generations",
            "getGeneration",
            ...queryKeyBase,
        ],
    });
}
export function invalidateAllGenerationsGetGeneration(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Generations", "getGeneration"],
    });
}
export function buildGenerationsGetGenerationQuery(client$, request, options) {
    return {
        queryKey: queryKeyGenerationsGetGeneration({ id: request.id }),
        queryFn: async function generationsGetGenerationQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(generationsGetGeneration(client$, request, mergedOptions));
        },
    };
}
export function queryKeyGenerationsGetGeneration(parameters) {
    return ["@openrouter/sdk", "Generations", "getGeneration", parameters];
}
