// @openrouter/sdk/esm/react-query/modelsCount.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { modelsCount } from "../funcs/modelsCount.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Get total count of available models
 */
export function useModelsCount(options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildModelsCountQuery(client, options),
        ...options,
    });
}
/**
 * Get total count of available models
 */
export function useModelsCountSuspense(options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildModelsCountQuery(client, options),
        ...options,
    });
}
export function prefetchModelsCount(queryClient, client$) {
    return queryClient.prefetchQuery({
        ...buildModelsCountQuery(client$),
    });
}
export function setModelsCountData(client, data) {
    const key = queryKeyModelsCount();
    return client.setQueryData(key, data);
}
export function invalidateAllModelsCount(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Models", "count"],
    });
}
export function buildModelsCountQuery(client$, options) {
    return {
        queryKey: queryKeyModelsCount(),
        queryFn: async function modelsCountQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(modelsCount(client$, mergedOptions));
        },
    };
}
export function queryKeyModelsCount() {
    return ["@openrouter/sdk", "Models", "count"];
}
