// @openrouter/sdk/esm/react-query/modelsListForUser.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { modelsListForUser } from "../funcs/modelsListForUser.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * List models filtered by user provider preferences
 */
export function useModelsListForUser(security, options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildModelsListForUserQuery(client, security, options),
        ...options,
    });
}
/**
 * List models filtered by user provider preferences
 */
export function useModelsListForUserSuspense(security, options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildModelsListForUserQuery(client, security, options),
        ...options,
    });
}
export function prefetchModelsListForUser(queryClient, client$, security) {
    return queryClient.prefetchQuery({
        ...buildModelsListForUserQuery(client$, security),
    });
}
export function setModelsListForUserData(client, data) {
    const key = queryKeyModelsListForUser();
    return client.setQueryData(key, data);
}
export function invalidateAllModelsListForUser(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Models", "listForUser"],
    });
}
export function buildModelsListForUserQuery(client$, security, options) {
    return {
        queryKey: queryKeyModelsListForUser(),
        queryFn: async function modelsListForUserQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(modelsListForUser(client$, security, mergedOptions));
        },
    };
}
export function queryKeyModelsListForUser() {
    return ["@openrouter/sdk", "Models", "listForUser"];
}
