// @openrouter/sdk/esm/react-query/modelsList.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { modelsList } from "../funcs/modelsList.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * List all models and their properties
 */
export function useModelsList(request, options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildModelsListQuery(client, request, options),
        ...options,
    });
}
/**
 * List all models and their properties
 */
export function useModelsListSuspense(request, options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildModelsListQuery(client, request, options),
        ...options,
    });
}
export function prefetchModelsList(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildModelsListQuery(client$, request),
    });
}
export function setModelsListData(client, queryKeyBase, data) {
    const key = queryKeyModelsList(...queryKeyBase);
    return client.setQueryData(key, data);
}
export function invalidateModelsList(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Models", "list", ...queryKeyBase],
    });
}
export function invalidateAllModelsList(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Models", "list"],
    });
}
export function buildModelsListQuery(client$, request, options) {
    return {
        queryKey: queryKeyModelsList({
            category: request?.category,
            supportedParameters: request?.supportedParameters,
        }),
        queryFn: async function modelsListQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(modelsList(client$, request, mergedOptions));
        },
    };
}
export function queryKeyModelsList(parameters) {
    return ["@openrouter/sdk", "Models", "list", parameters];
}
