// @openrouter/sdk/esm/react-query/oAuthExchangeAuthCodeForAPIKey.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { oAuthExchangeAuthCodeForAPIKey } from "../funcs/oAuthExchangeAuthCodeForAPIKey.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Exchange authorization code for API key
 *
 * @remarks
 * Exchange an authorization code from the PKCE flow for a user-controlled API key
 */
export function useOAuthExchangeAuthCodeForAPIKeyMutation(options) {
    const client = useOpenRouterContext();
    return useMutation({
        ...buildOAuthExchangeAuthCodeForAPIKeyMutation(client, options),
        ...options,
    });
}
export function mutationKeyOAuthExchangeAuthCodeForAPIKey() {
    return ["@openrouter/sdk", "OAuth", "exchangeAuthCodeForAPIKey"];
}
export function buildOAuthExchangeAuthCodeForAPIKeyMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyOAuthExchangeAuthCodeForAPIKey(),
        mutationFn: function oAuthExchangeAuthCodeForAPIKeyMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(oAuthExchangeAuthCodeForAPIKey(client$, request, mergedOptions));
        },
    };
}
