// @openrouter/sdk/esm/react-query/oAuthCreateAuthCode.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { oAuthCreateAuthCode } from "../funcs/oAuthCreateAuthCode.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Create authorization code
 *
 * @remarks
 * Create an authorization code for the PKCE flow to generate a user-controlled API key
 */
export function useOAuthCreateAuthCodeMutation(options) {
    const client = useOpenRouterContext();
    return useMutation({
        ...buildOAuthCreateAuthCodeMutation(client, options),
        ...options,
    });
}
export function mutationKeyOAuthCreateAuthCode() {
    return ["@openrouter/sdk", "OAuth", "createAuthCode"];
}
export function buildOAuthCreateAuthCodeMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyOAuthCreateAuthCode(),
        mutationFn: function oAuthCreateAuthCodeMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(oAuthCreateAuthCode(client$, request, mergedOptions));
        },
    };
}
