// @openrouter/sdk/esm/react-query/parametersGetParameters.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { parametersGetParameters } from "../funcs/parametersGetParameters.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * Get a model's supported parameters and data about which are most popular
 */
export function useParametersGetParameters(security, request, options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildParametersGetParametersQuery(client, security, request, options),
        ...options,
    });
}
/**
 * Get a model's supported parameters and data about which are most popular
 */
export function useParametersGetParametersSuspense(security, request, options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildParametersGetParametersQuery(client, security, request, options),
        ...options,
    });
}
export function prefetchParametersGetParameters(queryClient, client$, security, request) {
    return queryClient.prefetchQuery({
        ...buildParametersGetParametersQuery(client$, security, request),
    });
}
export function setParametersGetParametersData(client, queryKeyBase, data) {
    const key = queryKeyParametersGetParameters(...queryKeyBase);
    return client.setQueryData(key, data);
}
export function invalidateParametersGetParameters(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@openrouter/sdk",
            "Parameters",
            "getParameters",
            ...queryKeyBase,
        ],
    });
}
export function invalidateAllParametersGetParameters(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Parameters", "getParameters"],
    });
}
export function buildParametersGetParametersQuery(client$, security, request, options) {
    return {
        queryKey: queryKeyParametersGetParameters(request.author, request.slug, {
            provider: request.provider,
        }),
        queryFn: async function parametersGetParametersQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(parametersGetParameters(client$, security, request, mergedOptions));
        },
    };
}
export function queryKeyParametersGetParameters(author, slug, parameters) {
    return [
        "@openrouter/sdk",
        "Parameters",
        "getParameters",
        author,
        slug,
        parameters,
    ];
}
