// @openrouter/sdk/esm/react-query/providersList.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { providersList } from "../funcs/providersList.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * List all providers
 */
export function useProvidersList(options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildProvidersListQuery(client, options),
        ...options,
    });
}
/**
 * List all providers
 */
export function useProvidersListSuspense(options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildProvidersListQuery(client, options),
        ...options,
    });
}
export function prefetchProvidersList(queryClient, client$) {
    return queryClient.prefetchQuery({
        ...buildProvidersListQuery(client$),
    });
}
export function setProvidersListData(client, data) {
    const key = queryKeyProvidersList();
    return client.setQueryData(key, data);
}
export function invalidateAllProvidersList(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Providers", "list"],
    });
}
export function buildProvidersListQuery(client$, options) {
    return {
        queryKey: queryKeyProvidersList(),
        queryFn: async function providersListQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(providersList(client$, mergedOptions));
        },
    };
}
export function queryKeyProvidersList() {
    return ["@openrouter/sdk", "Providers", "list"];
}
